pkg/apis/networking/types.go

```diff
@@ -60,10 +71,18 @@ type NetworkPolicySpec struct {
 	// allowed if there are no NetworkPolicies selecting the pod (and cluster policy
 	// otherwise allows the traffic), OR if the traffic matches at least one egress rule
 	// across all of the NetworkPolicy objects whose podSelector matches the pod. If
-	// this field is empty then this NetworkPolicy does not limit any outgoing traffic
-	// (and serves solely to ensure that the pods it selects does not allow any outgoing traffic.)
+	// this field is empty then this NetworkPolicy does not limit any outgoing traffic.
+	// This field is beta-level in 1.8
 	// +optional
 	Egress []NetworkPolicyEgressRule
+
+	// List of rule types that the NetworkPolicy relates to.
+	// Valid options are Ingress, Egress, or Ingress,Egress. If this field is not specified, it will
+	// default based on the existence of Ingress or Egress rules. PolicyTypes will always include Ingress
+	// in the list unless the field only contains Egress.
```

Comment:
"PolicyTypes will always include Ingress in the list unless it doesn't" :-)

We need to be more explicit about the defaulting rules...

If this field is not specified, it will default based on the existence of Ingress or Egress rules; policies that contain an Egress section are assumed to affect Egress, and all policies (whether or not they contain an Ingress section) are assumed to affect Ingress. If you want to write an egress-only policy, you must explicitly specify PolicyTypes [ "Egress" ]. Likewise, if you want to write a policy the specifies that no egress is allowed, you must specify a PolicyTypes values that include "Egress" (since such a policy would not include an Egress section and would otherwise default to just [ "Ingress" ]).